core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -452,8 +452,12 @@ trait DerivationSchema[R] {
     getDirectives(ctx.annotations)
 
   private def getName(annotations: Seq[Any], typeName: TypeName): String =
-    annotations.collectFirst { case GQLName(name) => name }
-      .getOrElse(typeName.short + typeName.typeArguments.map(_.short).mkString)
+    annotations.collectFirst { case GQLName(name) => name }.getOrElse {
+      typeName.typeArguments match {
+        case Nil  => typeName.short
+        case args => typeName.short + args.map(getName(annotations, _)).mkString
```

Comment:
I think you can pass an empty list for annotations, to avoid searching something that won’t be found.

Reply:
good point 👍 updated.